python/dask_cudf/dask_cudf/core.py

```diff
@@ -374,62 +374,9 @@ def last_index(df):
         divisions = compute(*divs)
         return type(self)(self.dask, self._name, self._meta, divisions)
 
-    def set_index(self, index, drop=True, sorted=False):
-        """Set new index.
-
-        Parameters
-        ----------
-        index : str or Series
-            If a ``str`` is provided, it is used as the name of the
-            column to be made into the index.
-            If a ``Series`` is provided, it is used as the new index
-        drop : bool
-            Whether the first original index column is dropped.
-        sorted : bool
-            Whether the new index column is already sorted.
-        """
-        if not drop:
-            raise NotImplementedError("drop=False not supported yet")
-
-        if isinstance(index, str):
-            tmpdf = self.sort_values(index)
-            return tmpdf._set_column_as_sorted_index(index, drop=drop)
-        elif isinstance(index, Series):
-            indexname = "__dask_cudf.index"
-            df = self.assign(**{indexname: index})
-            return df.set_index(indexname, drop=drop, sorted=sorted)
-        else:
-            raise TypeError("cannot set_index from {}".format(type(index)))
-
-    def _set_column_as_sorted_index(self, colname, drop):
-        def select_index(df, col):
-            return df.set_index(col)
-
-        return self.map_partitions(
-            select_index, col=colname, meta=self._meta.set_index(colname)
-        )
-
-    def _argsort(self, col, sorted=False):
-        """
-        Returns
-        -------
-        shufidx : Series
-            Positional indices to be used with .take() to
-            put the dataframe in order w.r.t ``col``.
-        """
-        # Get subset with just the index and positional value
-        subset = self[col].to_dask_dataframe()
-        subset = subset.reset_index(drop=False)
-        ordered = subset.set_index(0, sorted=sorted)
-        shufidx = from_dask_dataframe(ordered)["index"]
-        return shufidx
-
-    def _set_index_raw(self, indexname, drop, sorted):
-        shufidx = self._argsort(indexname, sorted=sorted)
-        # Shuffle the GPU data
-        shuffled = self.take(shufidx, npartitions=self.npartitions)
-        out = shuffled.map_partitions(lambda df: df.set_index(indexname))
-        return out
+    def set_index(self, other, **kwargs):
+        kwargs.pop("shuffle", None)
```

Comment:
This seems a little dangerous because if the user provides `shuffle="disk"` we silently ignore this.  I might instead raise an error here.

```suggestion
        if kwargs.get("shuffle", "tasks") != "tasks":
            raise ValueError("Dask-cudf only supports task based shuffling, got %s" % kwargs["shuffle"])
```

Reply:
Good point - we should definitely be explicit here.